TripKit/Classes/helpers/TKUserProfileHelper.swift

```diff
@@ -70,16 +76,39 @@ public class TKUserProfileHelper: NSObject {
   
   @objc public class func setModeIdentifier(_ modeIdentifier: Identifier, toHidden hidden: Bool) {
     update(hiddenModeIdentifiers, forKey: .hidden, modeIdentifier: modeIdentifier, include: hidden)
+    
+    // We give special consideration when modifying walking or wheelchair mode since we may
+    // need to perform toggle, e.g., adding walking while wheelchair is already included
+    // calls for toggle.
+    let isModifyingWalking = modeIdentifier == SVKTransportModeIdentifierWalking
+    let walkingIsCurrentlyIncluded = !modeIdentifierIsHidden(SVKTransportModeIdentifierWalking)
+    let isModifyingWheelchair = modeIdentifier == SVKTransportModeIdentifierWheelchair
+    let wheelchairIsCurrentlyIncluded = !modeIdentifierIsHidden(SVKTransportModeIdentifierWheelchair)
+    
+    if isModifyingWalking && wheelchairIsCurrentlyIncluded {
+      update(hiddenModeIdentifiers, forKey: .hidden, modeIdentifier: SVKTransportModeIdentifierWheelchair, include: !hidden)
+    } else if isModifyingWheelchair && walkingIsCurrentlyIncluded {
+      update(hiddenModeIdentifiers, forKey: .hidden, modeIdentifier: SVKTransportModeIdentifierWalking, include: !hidden)
+    }
   }
   
   private class func update(_ identifiers: Set<Identifier>, forKey key: DefaultsKey, modeIdentifier: Identifier, include: Bool) {
     var modes = identifiers
+    
     if include {
       modes.insert(modeIdentifier)
     } else {
       modes.remove(modeIdentifier)
     }
+    
     UserDefaults.shared.set(Array(modes), forKey: key.rawValue)
+    
+    // Whenever we update the list of hidden modes, we need to consider if
+    // we are modifying wheelchair mode as it impacts whether we show
+    // wheelchair info.
+    if key == .hidden && modeIdentifier == SVKTransportModeIdentifierWheelchair {
+      showWheelchairInformation = !include
+    }
```

Comment:
Please add a test for this logic, too.

Reply:
Feel like this is missing a few cases: What if you show wheelchair again? What should happen if you hide or unhindered walking

Reply:
This has since been removed.